k9mail/src/main/java/com/fsck/k9/activity/setup/autoconfiguration/AccountSetupAutoConfigurationPresenter.java

```diff
@@ -0,0 +1,576 @@
+                    String transportURIHost = (Uri.parse(account.getTransportUri())).getHost();
+
+                    for (List<?> subjectAlternativeName : subjectAlternativeNames) {
+                        Integer type = (Integer)subjectAlternativeName.get(0);
+                        Object value = subjectAlternativeName.get(1);
+                        String name;
+                        switch (type.intValue()) {
+                            case 0:
+                                Timber.w("SubjectAltName of type OtherName not supported.");
+                                continue;
+                            case 1: // RFC822Name
+                                name = (String)value;
+                                break;
+                            case 2:  // DNSName
+                                name = (String)value;
+                                break;
+                            case 3:
+                                Timber.w("unsupported SubjectAltName of type x400Address");
+                                continue;
+                            case 4:
+                                Timber.w("unsupported SubjectAltName of type directoryName");
+                                continue;
+                            case 5:
+                                Timber.w("unsupported SubjectAltName of type ediPartyName");
+                                continue;
+                            case 6:  // Uri
+                                name = (String)value;
+                                break;
+                            case 7: // ip-address
+                                name = (String)value;
+                                break;
+                            default:
+                                Timber.w("unsupported SubjectAltName of unknown type");
+                                continue;
+                        }
+
+                        // if some of the SubjectAltNames match the store or transport -host,
+                        // display them
+                        if (name.equalsIgnoreCase(storeURIHost) || name.equalsIgnoreCase(transportURIHost)) {
+                            //TODO: localize this string
+                            altNamesText.append("Subject(alt): ").append(name).append(",...\n");
+                        } else if (name.startsWith("*.") && (
+                                storeURIHost.endsWith(name.substring(2)) ||
+                                        transportURIHost.endsWith(name.substring(2)))) {
+                            //TODO: localize this string
+                            altNamesText.append("Subject(alt): ").append(name).append(",...\n");
+                        }
+                    }
+                    chainInfo.append(altNamesText);
+                }
+            } catch (Exception e1) {
+                // don't fail just because of subjectAltNames
+                Timber.w(e1, "cannot display SubjectAltNames in dialog");
```

Comment:
Thanks for your review! These codes are copied from `AccountSetupCheckSettings` and not written by me. But I will fix these issues.